addons/hr_expense/models/account_move.py

```diff
@@ -1,77 +1,93 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
+from markupsafe import Markup
 
-from odoo import models, fields, api, _
-from odoo.api import ondelete
-from odoo.exceptions import UserError
+from odoo import Command, models, fields, api, _
+from odoo.exceptions import ValidationError
 from odoo.tools.misc import frozendict
 
 
 class AccountMove(models.Model):
     _inherit = "account.move"
 
-    expense_sheet_id = fields.Many2one(comodel_name='hr.expense.sheet', ondelete='set null', copy=False, index='btree_not_null')
-    show_commercial_partner_warning = fields.Boolean(compute='_compute_show_commercial_partner_warning')
+    expense_ids = fields.One2many(comodel_name='hr.expense', inverse_name='account_move_id')
+    nb_expenses = fields.Integer(compute='_compute_nb_expenses', string='Number of Expenses', compute_sudo=True)
 
-    @api.depends('partner_id', 'expense_sheet_id', 'company_id')
+    def _compute_nb_expenses(self):
+        for move in self:
+            move.nb_expenses = len(move.expense_ids)
+
+    @api.depends('partner_id', 'expense_ids', 'company_id')
     def _compute_commercial_partner_id(self):
-        own_expense_moves = self.filtered(lambda move: move.sudo().expense_sheet_id.payment_mode == 'own_account')
+        own_expense_moves = self.filtered(lambda move: any(expense.payment_mode == 'own_account' for expense in move.sudo().expense_ids))
         for move in own_expense_moves:
-            if move.expense_sheet_id.payment_mode == 'own_account':
-                move.commercial_partner_id = (
-                    move.partner_id.commercial_partner_id
-                    if move.partner_id.commercial_partner_id != move.company_id.partner_id
-                    else move.partner_id
-                )
+            move.commercial_partner_id = (
+                move.partner_id.commercial_partner_id
+                if move.partner_id.commercial_partner_id != move.company_id.partner_id
+                else move.partner_id
+            )
         super(AccountMove, self - own_expense_moves)._compute_commercial_partner_id()
 
-    def action_open_expense_report(self):
+    @api.constrains('expense_ids')
+    def _check_expense_ids(self):
+        for move in self:
+            expense_payment_modes = move.expense_ids.mapped('payment_mode')
+            if 'company_account' in expense_payment_modes and len(move.expense_ids) > 1 :
+                raise ValidationError(_("Only one journal entry can be linked per expense, if the expense was paid by the company"))
```

Comment:
not super clear: the error message should be inversed, IMO. Maybe @Laura-labe can suggest something else? maybe
```suggestion
                raise ValidationError(_("An expense paid by the company must have its dedicated journal entry"))
```

Reply:
I would show "Each expense paid by the company must have a distinct and dedicated journal entry." @qdp-odoo @JulienAlardot